File: api-tester/src/defaults/java/com/revenuecat/apitester/java/revenuecatui/CustomerCenterViewAPI.java
package com.revenuecat.apitester.java.revenuecatui;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.compose.ui.platform.AbstractComposeView;

import com.revenuecat.purchases.customercenter.CustomerCenterListener;
import com.revenuecat.purchases.ui.revenuecatui.views.CustomerCenterView;
import kotlin.Unit;
import kotlin.jvm.functions.Function0;

@SuppressWarnings({"unused"})
final class CustomerCenterViewAPI {

    static void checkType(@NonNull Context context) {
        AbstractComposeView view = new CustomerCenterView(context);
    }

    static void checkConstructors(
            @NonNull Context context,
            @Nullable AttributeSet attrs,
            int defStyleAttr,
            @NonNull Runnable dismissHandler,
            @NonNull CustomerCenterListener listener
    ) {
        Function0<Unit> dismissHandlerFunction = new Function0<Unit>() {
            @Override
            public Unit invoke() {
                dismissHandler.run();
                return Unit.INSTANCE;
            }
        };

        new CustomerCenterView(context);
        new CustomerCenterView(context, attrs);
        new CustomerCenterView(context, attrs, defStyleAttr);
        new CustomerCenterView(context, dismissHandlerFunction);
        new CustomerCenterView(context, listener);
        new CustomerCenterView(context, listener, dismissHandlerFunction);
    }

    static void checkMethods(
            @NonNull CustomerCenterView view,
            @NonNull Runnable dismissHandler,
            @NonNull CustomerCenterListener listener
    ) {
        Function0<Unit> dismissHandlerFunction = new Function0<Unit>() {
            @Override
            public Unit invoke() {
                dismissHandler.run();
                return Unit.INSTANCE;
            }
        };

        view.setDismissHandler(null);
        view.setDismissHandler(dismissHandlerFunction);
        view.setCustomerCenterListener(null);
        view.setCustomerCenterListener(listener);
    }
}

File: api-tester/src/defaults/kotlin/com/revenuecat/apitester/kotlin/revenuecatui/CustomerCenterViewAPI.kt
package com.revenuecat.apitester.kotlin.revenuecatui

import android.content.Context
import android.util.AttributeSet
import androidx.compose.ui.platform.AbstractComposeView
import com.revenuecat.purchases.customercenter.CustomerCenterListener
import com.revenuecat.purchases.ui.revenuecatui.views.CustomerCenterView

@Suppress("unused", "UNUSED_VARIABLE")
private class CustomerCenterViewAPI {

    fun checkType(context: Context) {
        val view: AbstractComposeView = CustomerCenterView(context)
    }

    fun checkConstructors(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        dismissHandler: () -> Unit,
        listener: CustomerCenterListener,
    ) {
        CustomerCenterView(context)
        CustomerCenterView(context, attrs)
        CustomerCenterView(context, attrs, defStyleAttr)
        CustomerCenterView(context, dismissHandler)
        CustomerCenterView(context, dismissHandler, listener)
    }

    fun checkMethods(
        view: CustomerCenterView,
        dismissHandler: () -> Unit,
        listener: CustomerCenterListener,
    ) {
        view.setDismissHandler(null)
        view.setDismissHandler(dismissHandler)
        view.setCustomerCenterListener(null)
        view.setCustomerCenterListener(listener)
    }
}
